Sources/SPMBuildCore/BinaryTarget+Extensions.swift

```diff
@@ -43,8 +43,8 @@ extension BinaryTarget {
         // Filter the libraries that are relevant to the triple.
         // FIXME: this filter needs to become more sophisticated
         guard let library = metadata.libraries.first(where: {
-            $0.platform == triple.os.asXCFrameworkPlatformString &&
-                $0.architectures.contains(triple.arch.rawValue)
+            $0.platform == triple.os?.asXCFrameworkPlatformString &&
+                $0.architectures.contains(triple.archName)
```

Comment:
can architectures be made more type safe instead of string array?

Reply:
I have absolutely no idea how this XCFramework code works and I'd like to avoid touching it if possible in this PR.
```
struct XCFrameworkMetadata.Library {
    let libraryIdentifier: String
    let libraryPath: String
    let headersPath: String?
    let platform: String
    let architectures: [String]
}
```